src/root/components/ThreeWForm/index.tsx

```diff
@@ -0,0 +1,631 @@
+    nationalSocietyOptions,
+    countryOptions,
+    districtOptions,
+    sectorOptions,
+    secondarySectorOptions,
+    programmeTypeOptions,
+    operationTypeOptions,
+    currentOperationOptions,
+    currentEmergencyOperationOptions,
+    operationToDisasterMap,
+    projectVisibilityOptions,
+    shouldDisableDistrictInput,
+    shouldShowCurrentEmergencyOperation,
+    shouldShowCurrentOperation,
+    districtPlaceholder,
+    disasterTypeOptions,
+    currentOperationPlaceholder,
+    disasterTypePlaceholder,
+    shouldDisableDisasterType,
+    statuses,
+    isReachedTotalRequired,
+    shouldDisableTotalTarget,
+    shouldDisableTotalReached,
+  } = useThreeWOptions(value);
+
+
+  // Calculate and set project status
+  React.useEffect(() => {
+    if (value.is_project_completed) {
+      onValueChange(PROJECT_STATUS_COMPLETED, 'status');
+      return;
+    }
+
+    if (!isDefined(value.start_date)) {
+      onValueChange(undefined, 'status');
+      return;
+    }
+
+    const startDate = new Date(value.start_date);
+    const now = new Date();
+
+    if (startDate <= now) {
+      onValueChange(PROJECT_STATUS_ONGOING, 'status');
+      return;
+    }
+
+    onValueChange(PROJECT_STATUS_PLANNED, 'status');
+  }, [onValueChange, value.start_date, value.is_project_completed]);
+
+  // Calculate and set target total
+  React.useEffect(() => {
+    if (isFalsy(value.target_male)
+      && isFalsy(value.target_female)
+      && isFalsy(value.target_other)
+    ) {
+      return;
+    }
+
+    const total = (value.target_male ?? 0)
+      + (value.target_female ?? 0)
+      + (value.target_other ?? 0);
+    onValueChange(total, 'target_total');
+  }, [onValueChange, value.target_male, value.target_female, value.target_other]);
+
+  // Calculate and set reached total
+  React.useEffect(() => {
+    if (isFalsy(value.reached_male)
+      && isFalsy(value.reached_female)
+      && isFalsy(value.reached_other)
+    ) {
+      return;
+    }
+
+    const total = (value.reached_male ?? 0)
+      + (value.reached_female ?? 0)
+      + (value.reached_other ?? 0);
+    onValueChange(total, 'reached_total');
+  }, [onValueChange, value.reached_male, value.reached_female, value.reached_other]);
+
+  // Set disaster type based on the selected event
+  React.useEffect(() => {
+    if (shouldDisableDisasterType) {
+      if (isDefined(value.event)) {
+        onValueChange(operationToDisasterMap[value.event], 'dtype');
+      }
+    }
+  }, [onValueChange, value.event, operationToDisasterMap, shouldDisableDisasterType]);
+
+  React.useEffect(() => {
+    projectCountryRef.current = value.project_country;
+  }, [value.project_country]);
+
+  // Clear out any selected region when country is changed
+  React.useEffect(() => {
+    if (isDefined(projectCountryRef.current)) {
+      if (projectCountryRef.current !== value.project_country) {
+        onValueChange([], 'project_districts');
+      }
+    }
+  }, [onValueChange, value.project_country]);
```

Comment:
Why not just use onValueSet?

Reply:
We need to discuss about this use case

Reply:
One side-effect is that onValueSet will clear error and set pristine. We should have a way to change that behaviour